rclcpp/include/rclcpp/node_impl.hpp

```diff
@@ -368,6 +368,18 @@ void Node::register_param_change_callback(CallbackT && callback)
   this->parameters_callback_ = callback;
 }
 
+template<typename ParameterT>
+bool Node::get_parameter(const std::string & name, ParameterT & parameter) const
+{
+  std::lock_guard<std::mutex> lock(mutex_);
+
+  rclcpp::parameter::ParameterVariant parameter_variant(parameter);
+  bool result = get_parameter_(name, parameter_variant);
+  parameter = parameter_variant.get_value<ParameterT>();
```

Comment:
This should not throw if `result == false`
That enables use like this
```
double foo = 0.1;
if (!get_parameter("foo", foo) {
  set_parameter("foo", foo);
}
```
We do throw if get_value fails because of differing types.